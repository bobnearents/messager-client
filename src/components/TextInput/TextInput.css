.message-input {
  height:40px;
  width:50%;
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
}

.message-form {
  text-align: center;
  margin-bottom:10px;
  
}

.message-input, .message-submit-button {
  margin:0;
  padding:0;
  border:1px solid black;
}

.message-submit-button {
  height:42px;
  width:7%;
  background:white;
  border-top-right-radius: 10px;
  border-bottom-right-radius: 10px;
}

.message-submit-button:hover {
  background:lawngreen;
}