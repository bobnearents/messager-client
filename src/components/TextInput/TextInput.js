import React, { Component } from 'react';
import './TextInput.css';
import ApiService from '../../services/api-service';
import MessageContext from '../../context/message-context'

class TextInput extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  static contextType = MessageContext;

  handleSubmit = event => {
    event.preventDefault();
    const { text } = event.target;
    ApiService.postMessage(text.value, this.context.room_id);
    event.target.text.value = "";
  };

  render() {
    return (
      <form className = "message-form" onSubmit={this.handleSubmit}>
        <input type="text" name="text" className="message-input" />
        <button type="submit" className="message-submit-button">
          Enter
        </button>
      </form>
    );
  }
}
 
export default TextInput;