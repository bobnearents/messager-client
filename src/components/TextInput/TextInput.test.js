import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import TextInput from "./TextInput";

describe("<TextInput />", () => {
  it("Renders without crashing", () => {
    shallow(<TextInput />);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<TextInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
