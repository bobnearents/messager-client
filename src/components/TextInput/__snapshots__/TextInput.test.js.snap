// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> renders correctly 1`] = `
<form
  className="message-form"
  onSubmit={[Function]}
>
  <input
    className="message-input"
    name="text"
    type="text"
  />
  <button
    className="message-submit-button"
    type="submit"
  >
    Enter
  </button>
</form>
`;
